core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -460,8 +460,8 @@ trait WhiskWebActionsApi extends Directives with ValidateRequestSize with PostAc
    * This method is factored out to allow mock testing.
    */
   protected def getIdentity(namespace: EntityName)(implicit transid: TransactionId): Future[Identity] = {
-
-    authenticationProvider.identityByNamespace(namespace)(transid, authStore)
+    // ask auth provider to create an identity for the given namespace
+    authenticationProvider.identityByNamespace(namespace)(transid, actorSystem, authStore)
```

Comment:
I think it would have made more sense to put the actorSystem in the Trait; it doesn't seem used in core/controller/src/main/scala/whisk/core/controller/BasicAuthenticationDirective.scala and might be specific to the SPI... is that right?

Reply:
Which trait? `AuthenticationDirectiveProvider`? How would that've worked? Pass it in as a constructor parameter?